module/zfs/vdev.c

```diff
@@ -427,32 +427,53 @@ vdev_get_nparity(vdev_t *vd)
 }
 
 static int
-vdev_prop_get_int(vdev_t *vd, vdev_prop_t prop, uint64_t *value)
+vdev_prop_get_objid(vdev_t *vd, uint64_t *objid)
 {
-	spa_t *spa = vd->vdev_spa;
-	objset_t *mos = spa->spa_meta_objset;
-	uint64_t objid;
-	int err;
 
 	if (vd->vdev_root_zap != 0) {
-		objid = vd->vdev_root_zap;
+		*objid = vd->vdev_root_zap;
 	} else if (vd->vdev_top_zap != 0) {
-		objid = vd->vdev_top_zap;
+		*objid = vd->vdev_top_zap;
 	} else if (vd->vdev_leaf_zap != 0) {
-		objid = vd->vdev_leaf_zap;
+		*objid = vd->vdev_leaf_zap;
 	} else {
 		return (EINVAL);
 	}
 
+	return (0);
+}
+
+static int
+vdev_prop_get_int(vdev_t *vd, vdev_prop_t prop, uint64_t *value)
+{
+	spa_t *spa = vd->vdev_spa;
+	objset_t *mos = spa->spa_meta_objset;
+	uint64_t objid;
+	int err;
+
+	if (vdev_prop_get_objid(vd, &objid) != 0)
+		return (EINVAL);
+
 	err = zap_lookup(mos, objid, vdev_prop_to_name(prop),
 	    sizeof (uint64_t), 1, value);
-
 	if (err == ENOENT)
 		*value = vdev_prop_default_numeric(prop);
 
 	return (err);
 }
 
+static int
+vdev_prop_get_bool(vdev_t *vd, vdev_prop_t prop, boolean_t *bvalue)
+{
+	int err;
+	uint64_t ivalue;
+
+	err = vdev_prop_get_int(vd, prop, &ivalue);
+	*bvalue = ivalue & 1;
```

Comment:
Should it be ` & 1` or ` != 0`?

Reply:
But you ended up with ` != 1`? What's that?

Reply:
Combination of both.
To fast commit...